VirtoCommerce.PricingModule.Data/Extensions/IQueryableExtension.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using VirtoCommerce.Domain.Pricing.Model.Search;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.PricingModule.Data.Model;
+using coreModel = VirtoCommerce.Domain.Pricing.Model;
+
+namespace VirtoCommerce.PricingModule.Data.Extensions
+{
+    public static class IQueryableExtension
+    {
+        public static IQueryable<PricelistAssignmentEntity> BuildSearchQuery(this IQueryable<PricelistAssignmentEntity> query, PricelistAssignmentsSearchCriteria criteria)
+        {
+            query = GetQueryWithoutSkipTake(query, criteria);
+            query = query.Skip(criteria.Skip).Take(criteria.Take);
+
+            return query;
+        }
+
+        public static IQueryable<PricelistAssignmentEntity> BuildSearchQuery(this IQueryable<PricelistAssignmentEntity> query, PricelistAssignmentsSearchCriteria criteria, out int count)
```

Comment:
I think we can leave only one method, which would be without Count() and Skip.Take at all:
public static IQueryable<PricelistAssignmentEntity> BuildSearchQuery(this IQueryable<PricelistAssignmentEntity> query, PricelistAssignmentsSearchCriteria criteria)

If we need count - we call .Count() on built search criteria after extension call, if we need Skip.Take - we call it (in IPricingSearchService).